api/tests/unit/domain/read-models/participant-results/AssessmentResult_test.js

```diff
@@ -226,4 +230,73 @@ describe('Unit | Domain | Read-Models | ParticipantResult | AssessmentResult', (
       });
     });
   });
+
+  describe('#canImprove', () => {
+    const originalConstantValue = constants.MINIMUM_DELAY_IN_DAYS_BEFORE_IMPROVING;
+    const assessmentCreatedAt = new Date('2020-01-05T05:06:07Z');
+    let clock;
+
+    before(() => {
+      constants.MINIMUM_DELAY_IN_DAYS_BEFORE_IMPROVING = 4;
+    });
+
+    after(() => {
+      constants.MINIMUM_DELAY_IN_DAYS_BEFORE_IMPROVING = originalConstantValue;
+    });
+
+    beforeEach(() => {
+      clock = sinon.useFakeTimers(assessmentCreatedAt);
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
+
+    context('when the knowledge element has been created less than MINIMUM_DELAY_IN_DAYS_BEFORE_IMPROVING days before assessment was created', () => {
+      it('returns false', () => {
+        const ke = domainBuilder.buildKnowledgeElement({ status: KnowledgeElement.StatusType.INVALIDATED, createdAt: new Date('2020-01-03') });
+        const participationResults = {
+          knowledgeElements: [ke],
+          acquiredBadgeIds: [],
+          assessmentCreatedAt,
+        };
+        const targetProfile = { competences: [], stages: [], badges: [] };
+
+        const assessmentResult = new AssessmentResult(participationResults, targetProfile, false);
+
+        expect(assessmentResult.canImprove).to.be.false;
+      });
+    });
+
+    context('when the knowledge element is validated', () => {
+      it('returns false', () => {
+        const ke = domainBuilder.buildKnowledgeElement({ status: KnowledgeElement.StatusType.VALIDATED, createdAt: new Date('2020-01-01') });
+        const participationResults = {
+          knowledgeElements: [ke],
+          acquiredBadgeIds: [],
+          assessmentCreatedAt,
+        };
+        const targetProfile = { competences: [], stages: [], badges: [] };
+
+        const assessmentResult = new AssessmentResult(participationResults, targetProfile, false);
+
+        expect(assessmentResult.canImprove).to.be.false;
+      });
+    });
+
+    context('when the knowledge element is invalidated and has bee created more than MINIMUM_DELAY_IN_DAYS_BEFORE_RETRYING days before assessment was created', () => {
```

Comment:
Tu as oublié de changer le nom dans le 2eme test ^^
```suggestion
    context('when the knowledge element is invalidated and has bee created more than MINIMUM_DELAY_IN_DAYS_BEFORE_IMPROVING days before assessment was created', () => {
```